### PR TITLE
Update grpc.md

### PR DESCRIPTION
**Pull request by 836hardik-agrawal**
Add the documentation for api.proto and gadettracermanager.proto

# [Title: describe the change in one sentence]

Added the description about api.proto and gadgettracemanager.proto

## How to use

Documentation

## Testing done

Documentation

### CODE REVIEW

File: docs/api/grpc.md [hunk 1]
```diff
@@ -4,16 +4,110 @@ sidebar_position: 30
...
+## Enumerations
+1. Kind
+An enumeration that defines various data types that a field can have, such as Bool, Int8, Int16, String, etc.
+## Key Concepts
+`Gadgets`: Tools or components that perform specific functions within the system. They can be built-in or custom.\
+`Data Sources`: Gadgets produce data that is structured in fields within data sources.\
+`Parameters`: Gadgets can be configured with various parameters that control their behavior.\
+`Events`: Gadgets emit events during their operation, which can be monitored and processed by clients.
 
 ## gadgettracermanager.proto
```
Comment (flyth) [inline docs/api/grpc.md:59]: I think this whole section can be dropped (or moved to an internal developer section), as this is only being used internally.

File: docs/api/grpc.md [hunk 2]
```diff
@@ -4,16 +4,110 @@ sidebar_position: 30
...
+8. `GadgetDataArray`:
+Represents an array of DataElement objects, useful for collecting multiple pieces of data from a gadget.
+9. `Param`:
+Describes a parameter that can be used by a gadget. It includes a key, description, default value, type hint, and other metadata.
+10. `GadgetInfo`:
+Contains detailed information about a gadget, including its name, associated image, data sources, and parameters.
+11. `Datasource`:
+Describes a data source associated with a gadget, including its ID, name, type, fields, and other metadata.
+12. `Field`:
+Represents a field within a data source, including its name, offset, size, type, and various flags.
```
Comment (flyth) [inline docs/api/grpc.md:49]: This needs detailed information, but we can handle it in a follow-up PR.

**Review by flyth — COMMENTED** [review]
Thanks for working on this!

Could you please group everything by built-in / non built-in and make the non-built-in service/docs the default? Built-in gadgets are considered legacy and will be deprecated soon. By grouping it that way we can highlight and later on remove it easier.

Also, please preview the rendered output. Some list items require additional new-lines (or sub-lists) in order to be rendered correctly (example: Services -> BuiltInGadgetManager -> Calls.)

**Comment by 836hardik-agrawal** [comment]
> Thanks for working on this!
> 
> Could you please group everything by built-in / non built-in and make the non-built-in service/docs the default? Built-in gadgets are considered legacy and will be deprecated soon. By grouping it that way we can highlight and later on remove it easier.
> 
> Also, please preview the rendered output. Some list items require additional new-lines (or sub-lists) in order to be rendered correctly (example: Services -> BuiltInGadgetManager -> Calls.)

Thanks for suggestion!
I am not able to understand how to group built-in and non built-in .In each section should I place the non built services on top and built in at bottom ?

**Comment by flyth** [comment]
> > Thanks for working on this!
> > Could you please group everything by built-in / non built-in and make the non-built-in service/docs the default? Built-in gadgets are considered legacy and will be deprecated soon. By grouping it that way we can highlight and later on remove it easier.
> > Also, please preview the rendered output. Some list items require additional new-lines (or sub-lists) in order to be rendered correctly (example: Services -> BuiltInGadgetManager -> Calls.)
> 
> Thanks for suggestion! I am not able to understand how to group built-in and non built-in .In each section should I place the non built services on top and built in at bottom ?

Yeah, exactly.

I think the document should have two big sections:
* gRPC API (doesn't necessarily need this header)
* Legacy gRPC API for Built-In Gadgets

And inside both of them their corresponding parts of the API should be explained.

**Pull request closed without merging**